### PR TITLE
Don't include nan values in computed geometries

### PR DESCRIPTION
**Pull request by kwilcox**

### CODE REVIEW

**Comment by kwilcox** [comment]
`math.nan` not available prior to py3.5, fix incoming.

File: pocean/dsg/profile/im.py [hunk 1]
```diff
@@ -166,7 +167,8 @@ def calculated_metadata(self, df=None, geometries=True, clean_cols=True, clean_r
         first_row = df.iloc[0]
         first_loc = Point(first_row.x, first_row.y)
         if geometries:
-            coords = list(unique_justseen(zip(df.x, df.y)))
+            coords = unique_justseen(zip(df.x.tolist(), df.y.tolist()))
```
Comment (ocefpaf) [inline pocean/dsg/profile/im.py]: @kwilcox I guess you can simplify this using `df.x.dropna()` and avoid the list comprehension below, no?

Comment (kwilcox) [inline pocean/dsg/profile/im.py]: No because they the list lengths could potentially be of different sizes and I'd like to keep them the same.

Comment (ocefpaf) [inline pocean/dsg/profile/im.py]: I believe we can still avoid the loop be using `pd.isnull` to create a mask and then merging the masks.

Comment (kwilcox) [inline pocean/dsg/profile/im.py]: At the cost of code readability. Build for comfort, optimize when needed. You can make a PR but I'm not going to change this atm.

Comment (kwilcox) [inline pocean/dsg/profile/im.py]: @ocefpaf  I'm going to go ahead and contradict myself. Just tested on a huge glider deployment.

How's this? Passes the tests.

```python
# only extract non-null pairs
null_coordinates = tgroup.x.isnull() | tgroup.y.isnull()
coords = list(unique_justseen(zip(tgroup.x[~null_coordinates].tolist(), tgroup.y[~null_coordinates].tolist())))
```

Comment (ocefpaf) [inline pocean/dsg/profile/im.py]: >  I'm going to go ahead and contradict myself.

You did not contradict yourself. Only the need to optimize came sooner than expected :smile: 

LGTM and readable enough. Do we need to `tolist()`?

Comment (kwilcox) [inline pocean/dsg/profile/im.py]: Without `tolist()` there are floating point rounding errors

Comment (ocefpaf) [inline pocean/dsg/profile/im.py]: Got it. Thanks!

**Comment by kwilcox** [comment]
This 3.5 error in travis makes me very sad. Didn't this happen once before?

**Pull request merged**